packages/amplify-cli/src/extensions/amplify-helpers/resource-status.ts

```diff
@@ -7,17 +7,29 @@ import { hashElement, HashElementOptions } from 'folder-hash';
 import { getEnvInfo } from './get-env-info';
 import { CLOUD_INITIALIZED, CLOUD_NOT_INITIALIZED, getCloudInitStatus } from './get-cloud-init-status';
 import { ServiceName as FunctionServiceName, hashLayerResource } from 'amplify-category-function';
+import { hashGQLResource } from 'amplify-provider-awscloudformation';
 import { removeGetUserEndpoints } from '../amplify-helpers/remove-pinpoint-policy';
 import { pathManager, stateManager, $TSMeta, $TSAny, NotInitializedError } from 'amplify-cli-core';
 
-async function isBackendDirModifiedSinceLastPush(resourceName, category, lastPushTimeStamp, hashFunction) {
+async function isBackendDirModifiedSinceLastPush(resourceName: string, categoryName: string, meta: $TSMeta) {
```

Comment:
this logic was moved out for a reason in the layer's PR as it is not a dependent functions's role to have logic to decide about what hashing function to use, burying this logic inside will lead to hard to maintain+understand code, please move it out.